include/bluetooth/iso.h

```diff
@@ -362,6 +371,19 @@ int bt_iso_chan_disconnect(struct bt_iso_chan *chan);
  */
 int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf);
 
+/**
+ * @brief Get the metadata for a received ISO packet
+ *
+ * Received ISO data carry metadata such as ISO sequence  number and optionally
+ * a timestamp.
+ *
+ * @param buf pointer to a valid ISO buffer received through the recv callback.
+ *
+ * @return Pointer to the metadata for the buffer. The lifetime of the pointer
+ *         is linked to the lifetime of the net_buf.
+ */
+struct iso_meta_data *bt_iso_meta_data_get(struct net_buf *buf);
+
```

Comment:
Hmm, since we are doing ISO API changes: Does it make more sense to just add a pointer to a `struct iso_meta_data` entry in the `struct bt_iso_chan_ops` `recv` callback instead?

Reply:
@Vudentz do you agree? 
I like the idea, but wonder if there is a reason this was not already done?

Reply:
Or we don't strip the header which means the buffer would contain both the timestamp and sequence number as part of the data itself, that way we don't need to complicate either the user_data or reserver more space in the stack since the original buffer already contain the header.

Reply:
@Vudentz I simply extended your implementation of stripping off the timestamp and store it in user_data. I actually like that the application do not need to write/call a parser for the header data. The solution with storing header data in a separate array do not use more memory than extending user-data or extending each pool (except on 64bit systems, where it will use a bit more).
As we already use the separate-array approach for HCI, could we agree to stick to this solution and provide the header in the call-back as Emil suggests?

Reply:
Adding this to the receive callback will also following the convention of receive callbacks in bluetooth.h (i.e. an `info` object and the `data`).

Reply:
Done